### ISSUE TITLE
Allow the use of a virtual caret on any focused NVDAObject (e.g. MSN History)

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/test_nvdaTracImport#102]
**Reported by mdcurran on 24 Jan 2008 15:36 UTC**
**Blocking #25**
The History widget in MSN / Windows Live Messenger can get focus, but does not have an accessible caret. NVDA used to allow the arrows to move the review cursor in that object, however when the review scripts were moved to the default appModule this became impossible. The new virtualBuffer base class has a very good virtual caret, including the ability to select text -- something the review scripts do not offer. We should make these virtual caret scripts usable by not just virtualBuffers, but also NVDAObjects. It would still be up to the specific NVDAObject to bind to these scripts, but they would be a choice for all NVDAObjects. Firstly the scripts and helper functions should probably be renamed so that they clearly are talking about a virtualCaret, and then they should be moved in to some kind of virtualCaret class, which should be inherited from by NVDAObjects and virtualBuffers.

**Comment by michaelDCurran** [comment]
**Modified by jteh on 5 Feb 2008 07:30 UTC**

**Comment by michaelDCurran** [comment]
**Comment by jteh on 11 Feb 2008 04:45 UTC**
Implemented in r1667.

Rather than this support being part of the base NVDAObject, it is implemented as a mix-in class called !CursorManager. To use it, one must sub-class it (whether by single or multiple inheritance). This means it is only part of the object if required. See the new virtual buffer code or the MSN Messenger appModule for examples.

**Comment by michaelDCurran** [comment]
**Modified by jteh on 11 Feb 2008 04:46 UTC**
